script.js

```diff
@@ -1,4 +1,181 @@
+var reverseModeStatus = false;
+var userWinCount = 0;
+var computerWinCount = 0;
+var drawCount = 0;
+var userName = "";
+
+var RandomSPS = function () {
+  randomDecimal = Math.random();
+  randomInteger0to2 = Math.floor(randomDecimal * 3);
+  RandomInterger1to3 = randomInteger0to2 + 1;
+  if (RandomInterger1to3 == 1) {
+    computerChoiceMsg = "The computer chose Scissors!";
+    return "Scissors";
+  }
+  if (RandomInterger1to3 == 2) {
+    computerChoiceMsg = "The computer chose Paper!";
+    return "Paper";
+  }
+  if (RandomInterger1to3 == 3) {
+    computerChoiceMsg = "The computer chose Stone!";
+    return "Stone";
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  SPS = RandomSPS();
+
+  while (userName == "") {
+    userName = input;
+    if (userName == "") {
+      myOutputValue = "Please enter a valid username and click submit.";
+      return myOutputValue;
+    }
+    myOutputValue =
+      "Hi " +
+      userName +
+      ", play the game by entering either Scissors, Paper, or Stone and click submit!";
+    return myOutputValue;
+  }
+
+  if (input == "Reverse") {
```

Comment:
Would be great if you could put comments in your code :) 

For example. You could comment here // Check if user picked reverse mode 

Remember that as much as code is for the machine to run. It's also for people to read. Comments are important as you or someone else needs to be able to understand what each section is doing when reading through your code. 

Helpful tip: "If I see this code again in a few months will I understand what it does? "